digits/model/tasks/caffe_train.py

```diff
@@ -54,6 +54,14 @@ def upgrade_network(network):
         #TODO
         pass
 
+    @staticmethod
+    def set_mode(gpu):
+        if gpu is not None:
+            caffe.set_mode_gpu()
```

Comment:
I think you need to put `set_device()` BEFORE `set_mode_gpu()`. Odd.

https://groups.google.com/d/msg/caffe-users/HqdziMqpu6o/2zgy7MK3-hAJ
https://github.com/BVLC/caffe/pull/507